clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/model/EcsInstance.java

```diff
@@ -0,0 +1,82 @@
+package com.netflix.spinnaker.clouddriver.ecs.model;
+
+import com.amazonaws.services.ec2.model.InstanceStatus;
+import com.amazonaws.services.ecs.model.Task;
+import com.netflix.spinnaker.clouddriver.ecs.EcsCloudProvider;
+import com.netflix.spinnaker.clouddriver.model.HealthState;
+import com.netflix.spinnaker.clouddriver.model.Instance;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+public class EcsInstance implements Instance, Serializable {
+  private String name;
+  private HealthState healthState;
+  private Long launchTime;
+  private String zone;
+  private List<Map<String, String>> health;
+  private String providerType;
+  private String cloudProvider;
+
+  public EcsInstance(String name, Task task, InstanceStatus ec2Instance) {
+    this.name = name;
+    providerType = cloudProvider = EcsCloudProvider.ID;
+    launchTime = task.getStartedAt().getTime();
+    health =  null;
+    healthState = calculateHealthState(task.getLastStatus(), task.getDesiredStatus());
+    zone = ec2Instance.getAvailabilityZone();
+  }
+
+  @Override
```

Comment:
Style comment: I like putting simple getters and setters at the end of the class, so that when you read the class you get to the interesting stuff right at the beginning

Reply:
Updated